### PR TITLE
Bug/3217 allow nested flag component in header

### LINKED ISSUE #3217: [BUG] Kirby-header not allowing nested components
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
Kirby-header provides the functionality of adding a Kirby-flag, which is not working when the Kirby-flag is nested in a wrapper component. 

The same issue applies for avatar.

## Describe how to reproduce the bug
Create a new component which only holds a Kirby-flag and place it in the Kirby-header where the Kirby-flag should have been placed.

## Which Kirby version was used?
8.11.1 

## What was the expected behavior?
Kirby-header should display a nested Kirby-flag as if it was placed directly without the wrapper component.

## Please complete the following information:
- OS: MacOS
- Browser: Chrome
- Browser version: 116.0.5845.140 (Official Build) (arm64)

## Are there any additional context?
The problem resides in the header.component, and is two-fold. 

1) The header component has a ng-content that selects a Kirby-flag. 
2) The header component has a contentChild for KirbyFlagComponent, and a check for the child component being present in the DOM.

Issue 1 could be bypassed by using ngProjectAs on the nested flag component. However ngProjectAs does not work on contentChild, so due to the check for the content child being present, the nested component is still not shown.

This issue has been discussed with @jakobe.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/main/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

### PR DESCRIPTION
**Pull request by NillerW**
## Which issue does this PR close?

This PR closes #3217

## What is the new behavior?

By using ngProjectAs attribute on the a wrapper component, allows component nesting of avatar, kirby-progress-circle and flag in Kirby-header component.

The core functionality is still the same.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any additional context?

No tests has been added as existing tests covers the same functionality.

No cookbook examples has been added, since it should not be encouraged to make this bypass unless you know what you're doing - in agreement with @jakobe.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed] (Not relevant)

### CODE REVIEW

File: libs/designsystem/header/src/header.component.ts [hunk 1]
```diff
@@ -109,22 +116,19 @@ export class HeaderComponent implements AfterContentInit, OnInit {
...
+    const avatarPresent: boolean = this.avatarContainerElement?.nativeElement.children.length > 0;
+    if (avatarPresent && this.centered === undefined) {
       this.centered = true;
     }
 
     if (this.centered) {
       this.actionGroupConfig.isCondensed = true;
     }
+
+    this.avatarReadyToRenderAfterContentProjection = true;
```
Comment (NillerW) [inline libs/designsystem/header/src/header.component.ts]: Necessary due to the way kirby-progress-circle determines it's placement on the screen in relation to projected content.

Comment (NillerW) [reply libs/designsystem/header/src/header.component.ts]: Can no longer reproduce

File: libs/designsystem/calendar/src/calendar.component.scss [hunk 2]
```diff
@@ -96,6 +96,7 @@ td {
 .day.disabled,
 .day:not(.selectable) {
   --color: #{utils.get-text-color('semi-dark')};
+
```
Comment (jakobe) [inline libs/designsystem/calendar/src/calendar.component.scss]: Please revert unrelated change 🙏🏻  (it's a Prettier thing, but this file is not related to the issue at hand)
```suggestion
```

File: libs/designsystem/header/src/header.component.ts [hunk 3]
```diff
@@ -93,17 +84,31 @@ export class HeaderComponent implements AfterContentInit, OnInit {
     return Array.isArray(this.subtitle2) ? this.subtitle2 : [this.subtitle2];
   }
 
-  onTitleClick(event: PointerEvent) {
+  public onTitleClick(event: PointerEvent) {
```
Comment (jakobe) [inline libs/designsystem/header/src/header.component.ts]: Please revert to align with rest of codebase:
```suggestion
  onTitleClick(event: PointerEvent) {
```

File: libs/designsystem/header/src/header.component.ts [hunk 4]
```diff
@@ -93,17 +84,31 @@ export class HeaderComponent implements AfterContentInit, OnInit {
...
     this.titleClick.emit(event);
   }
 
-  _actionGroupInjector: Injector;
+  public _actionGroupInjector: Injector;
```
Comment (jakobe) [inline libs/designsystem/header/src/header.component.ts]: Please revert to align with rest of codebase:
```suggestion
 _actionGroupInjector: Injector;
```

File: libs/designsystem/header/src/header.component.ts [hunk 5]
```diff
@@ -93,17 +84,31 @@ export class HeaderComponent implements AfterContentInit, OnInit {
...
 
   private actionGroupConfig: ActionGroupConfig;
 
   constructor(private injector: Injector) {}
 
-  ngOnInit(): void {
+  public ngOnInit(): void {
```
Comment (jakobe) [inline libs/designsystem/header/src/header.component.ts]: Please revert to align with rest of codebase:
```suggestion
  ngOnInit(): void {
```

File: libs/designsystem/header/src/header.component.ts [hunk 6]
```diff
@@ -93,17 +84,31 @@ export class HeaderComponent implements AfterContentInit, OnInit {
...
+    this.initActionGroupInjector();
+
+    if (this.titleMaxLines > 0) {
+      this.fitHeadingConfig = {
+        maxLines: this.titleMaxLines,
+      };
+    }
+  }
+
+  public ngAfterContentChecked(): void {
```
Comment (jakobe) [inline libs/designsystem/header/src/header.component.ts]: Please align with rest of codebase:
```suggestion
  ngAfterContentChecked(): void {
```

File: libs/designsystem/header/src/header.component.ts [hunk 7]
```diff
@@ -117,17 +122,12 @@ export class HeaderComponent implements AfterContentInit, OnInit {
...
-        maxLines: this.titleMaxLines,
-      };
-    }
   }
 
-  ngAfterContentInit(): void {
+  private onPotentialAvatarRendered(): void {
     // If an avatar is present we default to centered layout - unless configured otherwise:
-    if (this.avatar && this.centered === undefined) {
+    const avatarPresent: boolean = this.avatarContainerElement?.nativeElement.children.length > 0;
```
Comment (jakobe) [inline libs/designsystem/header/src/header.component.ts:129]: Consider checking for visibility instead to better reflect the UI logic for the avatar container:
```suggestion
    const avatarPresent: boolean = this.avatarContainerElement?.nativeElement. checkVisibility();
```

File: libs/designsystem/header/src/header.component.ts (hunk 6 above)
Comment (jakobe) [inline libs/designsystem/header/src/header.component.ts]: Are we sure we want to change this from `ngAfterContentInit` to `ngAfterContentChecked `? 🤔 Is it needed for anything specific?

**Review by jakobe — CHANGES_REQUESTED** [review]
Looks good 👌🏻 

Only one thing to consider regarding life cycle change from `ngAfterContentInit` to  `ngAfterContentChecked`, see comment on specific lines.

Please also add a/some test(s) for the changes 🤞🏻 

Rest of comments is mainly around aligning code with rest of codebase 😇

**Comment by RasmusKjeldgaard** [comment]
Closing this, as we can look into this functionality when converting to web-components instead.
Please reopen if needed.

**Pull request closed without merging**